### PR TITLE
github action to check conflict in pr and add label

### LINKED ISSUE #13264: Dev: Bot to mark Pull Requests with "Has Conflicts"
It would be useful for a bot to check open pull requests for conflicts, and add a 'Has Conflicts' label if this is the case.

This label will be a useful metric to allow the stale bot identify old issues, and reduces the need for maintainers to manually check PRs for conflicts.

As an extension: this bot should remove the label once the conflicts have been resolved.

### PR DESCRIPTION
**Pull request by krmanik**
## Purpose / Description

Scan PR in current repo and add label to conflicting PR after other PR merged to main branch.

## Fixes
* Fixes #13264

## Approach
Using https://github.com/actions/github-script

Get all opened pr in current repo, get pr from list then get mergeable state of the pr, also added delay 1 second becuase on merge of other pr to main branch the mergeable state are unknown so after delay, the state may be dirty or clean, if dirty add `Has Conflicts`

## How Has This Been Tested?
Tested on temp repo

## Learning (optional, can help others)
https://octokit.github.io/rest.js/v20
https://github.com/actions/github-script

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: .github/workflows/conflict.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+name: Conflict Scan
+
+on:
+  schedule:
+  - cron: "* 12 * * *"
```
Comment (david-allison) [inline .github/workflows/conflict.yaml:5]: What's the intention here? This reads as 'once per minute between 12:00 and 13:00'

Ideally once per hour, open to other reviewers' thoughts (depending on how slo the job is)

File: .github/workflows/conflict.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                if (pullRequestList.data !== null && pullRequestList.data.length > 0) {
+                    for (let pullRequest of pullRequestList.data) {
+                        const prNumber = pullRequest.number;
+                        console.log(`Checking PR #${prNumber}`);
+
+                        let pullRequestData;
+                        do {
+                            pullRequestData = await getPullRequest(prNumber);
+                            // introduce 1-second delay before the next check
+                            await wait(1);
```
Comment (david-allison) [inline .github/workflows/conflict.yaml:71]: the parameter is `ms`, not `s`

File: .github/workflows/conflict.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                        } while (pullRequestData.data.mergeable_state === 'unknown');
```
Comment (david-allison) [inline .github/workflows/conflict.yaml:72]: Add a terminal condition or timeout somewhere - we don't want to spam GitHub

**Review by david-allison — CHANGES_REQUESTED** [review]
One question, one bug, one nitpick.

Intention looks great, thanks! Will really help with the maintenance burden

File: .github/workflows/conflict.yaml (hunk 1 above)
Comment (krmanik) [reply .github/workflows/conflict.yaml]: No intention, just visited this and copied https://crontab.guru/
Open for discussion, currently added 2 hour.

File: .github/workflows/conflict.yaml (hunk 3 above)
Comment (krmanik) [reply .github/workflows/conflict.yaml:80]: Added iteration count 10 or if state is not unknown

File: .github/workflows/conflict.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
+name: Conflict Scan
+
+on:
+  schedule:
+  - cron: "59 * * * *"
```
Comment (david-allison) [inline .github/workflows/conflict.yaml:5]: would be better at '0' IMO

**Review by david-allison — APPROVED** [review]
Thanks!

File: .github/workflows/conflict.yaml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+jobs:
+  scan_conflicts:
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (krmanik) [reply .github/workflows/conflict.yaml]: Not required, updated it.

File: .github/workflows/conflict.yaml (hunk 4 above)
Comment (krmanik) [reply .github/workflows/conflict.yaml]: done

**Review by lukstbit — APPROVED** [review]
LGTM. 
Please squash so we can merge this.

**Pull request merged**